src/jshint.js

```diff
@@ -4732,14 +4732,15 @@ var JSHINT = (function() {
       name = tkn.value;
     }
 
-    if (props[name] && _.has(props, name)) {
+    if ((props[name] && _.has(props, name)) || (name === "__proto__" &&
+        Object.getPrototypeOf(props) !== Object.prototype)) { // __proto__ has been modified
       warning("W075", state.tokens.next, msg, name);
-    } else {
-      props[name] = {};
     }
 
-    props[name].basic = true;
-    props[name].basictkn = tkn;
+    props[name] = {
```

Comment:
I think it might be safer if you didn't change it.. but I don't feel massively strongly.. wdyt?